lib/hooks/session/index.js

```diff
@@ -187,6 +187,29 @@ module.exports = function(app) {
       // Expose hook as `sails.session`
       app.session = SessionHook;
 
+      // If using Redis, wait for a connection or error
+      if (app.config.session.adapter === 'connect-redis' &&
+          // Future-proof by ensuring that there is a client object, and that it is an event emitter
```

Comment:
@sgress454 lgtm, just don't forget to use `app` (to avoid relying on the sails global)